### PR TITLE
Implement f{32,64}.neg instructions

### PR DESCRIPTION
**Pull request by gumb0**
image

### CODE REVIEW

File: test/unittests/execute_floating_point_test.cpp [hunk 1]
```diff
@@ -407,6 +407,28 @@ TYPED_TEST(execute_floating_point_types, abs)
     }
 }
 
+TYPED_TEST(execute_floating_point_types, neg)
```
Comment (chfast) [inline test/unittests/execute_floating_point_test.cpp:410]: Order the tests by opcode, if in the same fixture.

Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp]: Moved.

File: test/unittests/execute_floating_point_test.cpp [hunk 2]
```diff
@@ -337,6 +337,28 @@ TYPED_TEST(execute_floating_point_types, abs)
...
+        std::begin(this->positive_special_values), std::end(this->positive_special_values));
+    for (const auto x :
+        {TypeParam{0}, Limits::infinity(), FP::nan(FP::canon), FP::nan(FP::canon + 1), FP::nan(1)})
+        p_values.push_back(x);
+
+    for (const auto p : p_values)
+    {
+        // fneg(+-p) = -+p
+        EXPECT_THAT(exec(p), Result(-p));
+        EXPECT_THAT(exec(-p), Result(p));
```
Comment (axic) [inline test/unittests/execute_floating_point_test.cpp:358]: Aren't in these tests practically trusting the compiler compared to the others where we have explicit constants?

Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:358]: We trust that negation work everywhere.

Comment (axic) [reply test/unittests/execute_floating_point_test.cpp:358]: If we wanted to be pedantic we'd always test against the encoded representation.

Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:358]: Possible in the future... i.e. we would need to extend `FP` to do some software floating-point operations.

Comment (axic) [reply test/unittests/execute_floating_point_test.cpp:359]: Something like `wast2wasm4cpp` script could work, which evaluates the C++ code and puts in the hard constants. We could manually verify that when committed and at every change.

**Pull request merged**